README.md

```diff
@@ -0,0 +1,168 @@
+
+Follow the [documentation](http://docs.puppetlabs.com/guides/install_puppet/install_debian_ubuntu.html) to install Puppet on the node.
+
+For quick testing on Debian Wheezy, try the following:
+
+1. `wget https://apt.puppetlabs.com/puppetlabs-release-wheezy.deb`
+1. `sudo dpkg -i puppetlabs-release-wheezy.deb`
+1. `sudo apt-get update && sudo apt-get upgrade -y`
+1. `sudo apt-get install puppetmaster -y`
+
+For [Fog](fog.io/compute/) on Debian Wheezy install build-essential and ruby-dev with:
+
+1. `sudo apt-get build-essential ruby-dev -y`
+
+Install ruby gems `fog` and `google-api-client` with:
+
+1. `sudo gem install google-api-client --no-rdoc --no-ri`
+1. `sudo gem install fog --no-rdoc --no-ri`
+
+In order to use this repo as a Puppet module, create a symlink to Puppet's
+`modules` directory to your local copy of this git repo:
+
+1. `cd ~`
+1. `git clone https://github.com/GoogleCloudPlatform/gce_compute_fog gce_compute`
+1. `sudo ln -s ~/gce_compute /etc/puppet/modules/gce_compute`
+
+## Usage
+
+### Setting up authorization
+
+#### Auth with temporary tokens (recommended)
+
+This method requires a master with compute read/write permissions and will
+only work if your master is running in Google Compute Engine.  This auth
+method will retrieve an `authorization token` from GCE's internal metadata
+service.
+
+Create type called `gce_auth` with the name as the project id in your manifest. Your resource should look something like:
+```
+gce_auth { 'project-id':
+}
+```
+
+#### Auth using client email and key file
+
+If your master resides outside of Compute Engine, you will need to use a
+Service Account.
+
+1. Create a [GCE service account](https://developers.google.com/compute/docs/faq#howdoicreate). 
+2. Push the key file downloaded from step one to your master VM.
+3. Create a `gce_auth` resource in your manifest file that inculdes the client email and the absolute path to the key file. 
+```
+gce_auth { 'project-id':
+   client_email => <client email>
+   key_file     => </path/to/keyfile>
+}
+```
+
+#### For legacy manifest files (deprecated)
+
+The `--certname` option for `puppet apply` and `device.conf` can still be used *only if the master VM has compute read/write permissions*.
+
+### Making a manifest
+
+Once authentication is set up, the only thing left to do is make a manifest file.
+
+A simple manifest that creates 2 VMs is shown below.
+```
+gce_auth { 'project-id':
+}
+
+gce_disk { 'instance-1-disk':
+  ensure          => present,
+  source_image    => debian-7,
+  size_gb         => 10,
+  zone            => us-central1-a,
+}
+
+gce_instance { 'instance-1':
+  ensure          => present,
+  zone            => us-central1-a,
+  disks           => ['instance-1-disk,boot'],  
+  machine_type    => n1-standard-1,
+  puppet_master   => "$fqdn",
+  puppet_service  => present,
+  network         => 'default',
+}
+
+gce_disk { 'instance-2-disk':
```

Comment:
This could be collapsed into a single statement:

```puppet
gce_disk { ['instance-1-disk', 'instance-2-disk']:
  ensure             => present,
  source_image => debian-7,
  size_gb           => 10,
  zone                => us-central1-a,
}
```